flytekit/core/array_node.py

```diff
@@ -79,16 +79,23 @@ def __init__(
 
         # TODO - bound inputs are not supported at the moment
         self._bound_inputs: Set[str] = set()
+        self._excluded_inputs: Set[str] = set()
+        if isinstance(target, (LaunchPlan, FlyteLaunchPlan)):
```

Comment:
I do not think we need a nested if condition:

```suggestion
        if isinstance(target, (LaunchPlan, FlyteLaunchPlan)) and not isintance(target, ReferenceLaunchPlan):
```